library/src/main/java/com/otaliastudios/zoom/ZoomEngine.kt

```diff
@@ -1151,56 +1329,57 @@ internal constructor(context: Context) : ViewTreeObserver.OnGlobalLayoutListener
                                  allowOverScroll: Boolean) {
         if (setState(ANIMATING)) {
             mClearAnimation = false
-            val startTime = System.currentTimeMillis()
-            @ScaledPan val startX = scaledPanX
-            @ScaledPan val startY = scaledPanY
-            @ScaledPan val endX = startX + deltaX
-            @ScaledPan val endY = startY + deltaY
-            mContainer.post(object : Runnable {
-                override fun run() {
-                    if (mClearAnimation) return
-                    val time = interpolateAnimationTime(System.currentTimeMillis() - startTime)
-                    LOG.v("animateScaledPan:", "animationStep:", time)
-                    @ScaledPan val x = startX + time * (endX - startX)
-                    @ScaledPan val y = startY + time * (endY - startY)
-                    applyScaledPan(x - scaledPanX, y - scaledPanY, allowOverScroll)
-                    if (time >= 1f) {
-                        setState(NONE)
-                    } else {
-                        mContainer.postOnAnimation(this)
-                    }
+            val startPan = scaledPan
+            val endPan = startPan + ScaledPoint(deltaX, deltaY)
+
+            val panAnimator = ValueAnimator.ofObject(TypeEvaluator { fraction, startValue: ScaledPoint, endValue: ScaledPoint ->
+                startValue + (endValue - startValue) * fraction - scaledPan
+            }, startPan, endPan)
+            panAnimator.prepare()
+            panAnimator.addUpdateListener {
+                LOG.v("animateScaledPan:", "animationStep:", it.animatedFraction)
+                if (mClearAnimation) {
+                    it.cancel()
```

Comment:
Sorry I forgot this, we need a return after these it.cancel() to be as it was.

Reply:
right, I changed it in all 4 places.